test/minitest_helper.rb

```diff
@@ -158,10 +161,25 @@ def has_edge?(edge, traces)
       return true
     end
   end
-  AppOpticsAPM.logger.debug "[oboe/debug] edge #{edge} not found in traces."
+  AppOpticsAPM.logger.debug "[appoptics_apm/test] edge #{edge} not found in traces."
   false
 end
 
+def assert_entry_exit(traces, num = nil, check_task_id=true)
+  if check_task_id
+    task_id = traces[0]['X-Trace'][0..8]
+    refute traces.find{ |tr| tr['X-Trace'][0..8] != task_id }, "task ids not matching"
+  end
+  num_entries = traces.select { |tr| tr ['Label'] == 'entry'}.size
+  num_exits = traces.select { |tr| tr ['Label'] == 'exit'}.size
+  if num && num > 0
+    num_entries.must_equal num, "incorrect number of entry spans"
+    num_exits.must_equal num, "incorrect number of exit spans"
+  else
+    num_exits.must_equal num_entries, "number of exit spans is not the same as entry spans"
```

Comment:
I should maybe check this but don't have them all collected in one place to be checked.

Reply:
I started running into this problem because of the async nature of grpc_server and making sure there is exactly one exit event when exceptions happen.